src/Response.php

```diff
@@ -428,12 +506,32 @@ protected function appendHeaders(): self
 
         // Set application headers
         foreach ($this->headers as $key => $value) {
-            \header($key . ': ' . $value);
+            $this->sendHeader($key, $value);
         }
 
         return $this;
     }
 
+    protected function sendHeader(string $key, string $value)
+    {
+        \header($key . ': ' . $value);
+    }
+
+    protected function sendCookie(array $cookie) {
```

Comment:
let's keep the same signature as the php function.

Reply:
Missing docblock, what should be the return type?

Reply:
Signature looks different in different version?

Reply:
Yes, the changelog is available on the official docs: https://www.php.net/manual/en/function.setcookie.php. We can follow the latest one. Also use the null coalescing operator (??) to make sure we get all data properly.